guides/predictive-analytics-with-deepnote.md

```diff
@@ -0,0 +1,135 @@
+
+```sql
+SELECT * FROM public.salary LIMIT 100
+```
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Hydra dashboard.png" alt=""><figcaption></figcaption></figure>
+
+8. To access the table using Python scripts because predictions can not be done using SQL, an engine has to be created with the data warehouse credentials
+
+```python
+import pandas as pd
+from sqlalchemy import create_engine
+#user = username
+#pass = password
+#server = hostname
+#database = database-name
+engine = create_engine('postgresql://user:pass@server:5432/database')
+#table_name = Salary
+#database = public
+data = pd.read_sql_table('table_name',con= engine, schema= 'database')
+```
+
+**Congratulations!** You've successfully connected Hydra with Deepnote! What next?
+
+## Making the predictions
+
+Now that we have connected Deepnote to Hydra, we can use Python and SQL scripts to access the data. We will use the Linear Regression model to create predictions leveraging the salary data.
+
+The goal of linear regression model is to find a mathematical relationship or equation that best fits the data and helps us predict salaries based on the other factors.
+
+The necessary models and libraries are being imported.
+The linear regression model and matplotlib for our visualization are imported as shown below&#x20;
+
+```python
+#import Classes and lib needed
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+```
+
+The dataset is divided into dependent and independent variables(X, Y), in order for the linear regression model to understand the relationship between the varibles and make predictions.We train the linear regression model using the dataset. During the training process, the model learns from the data by identifying the patterns and relationships between the independent variables (years of experience) and the dependent variable (the salary).
+
+```python
+#defining variables X and Y
+X = data.iloc[:,0].values.reshape(-1,1)
+print(X)
+Y = data.iloc[:,1].values.reshape(-1,1)
+print(Y)
+```
+
+```python
+#creating a Linear Regression model for prediction
+model = LinearRegression()
+
+#Fitting the model on the Dataset
+model.fit(X,Y)
+
+Y_pred = model.predict(X)  # make predictions
+```
+
+Visualizations are created to show the difference between the actual salary versus the predicted salary.
+
+```python
+Y_pred = Y_pred.reshape(-1)
+difference = pd.DataFrame({'Real' : data. salary, 'Predicted': Y_pred, })
+print(difference)
+```
+
+```python
+graph = difference.head(10)
+graph.plot(kind='bar')
+plt.title('Actual vs Predicted')
+plt.ylabel('salary')
+```
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Output.png" alt="" width="464"><figcaption><p>Visualization</p></figcaption></figure>
+
+Your first prediction is complete! You can determine the range of a person's salary based on years of experience.
```

Comment:
One more small edit - You can predict the range of a person's salary based on years of experience.